proof-of-concept/pof.c

```diff
@@ -58,7 +58,7 @@ static HPy Point_new_impl (HPyContext ctx, HPy cls, HPy *args,
 HPyDef_SLOT(Point_repr, Point_repr_impl, HPy_tp_repr)
 static HPy Point_repr_impl(HPyContext ctx, HPy self)
 {
-    HPy_Point *point = HPy_CAST(ctx, HPy_Point, self);
+    HPy_Point *point = (HPy_Point*) HPy_Cast(ctx, self);
```

Comment:
shouldn't we define our own `HPy_Point_Cast`, which does the C-level casting under the hood?

Reply:
Done.